Source/WebCore/html/FormListedElement.cpp

```diff
@@ -165,20 +165,25 @@ void FormListedElement::formWillBeDestroyed()
 void FormListedElement::resetFormOwner()
 {
     RefPtr<HTMLFormElement> originalForm = form();
-    setForm(findAssociatedForm(&asHTMLElement(), originalForm.get()));
     HTMLElement& element = asHTMLElement();
+    setForm(findAssociatedForm(&element, originalForm.get()));
     auto* newForm = form();
     if (newForm && newForm != originalForm && newForm->isConnected())
         element.document().didAssociateFormControl(element);
 }
 
-void FormListedElement::formAttributeChanged()
+void FormListedElement::parseAttribute(const QualifiedName& name, const AtomString& value)
+{
+    if (name == formAttr)
+        parseFormAttribute(value);
+}
+
+void FormListedElement::parseFormAttribute(const AtomString& value)
 {
     HTMLElement& element = asHTMLElement();
-    if (!element.hasAttributeWithoutSynchronization(formAttr)) {
+    if (value.isNull()) {
         // The form attribute removed. We need to reset form owner here.
         RefPtr originalForm = form();
-        // FIXME: Why does this not pass originalForm to findClosestFormAncestor?
```

Comment:
What's the explanation? Maybe we can add a comment here if you know the reason.

Reply:
Added explanation in [d603dd0](https://github.com/WebKit/WebKit/pull/4988/commits/d603dd09bb92c34777c40fffbd8a53a7443fe453). This bit is covered by tests that fail if we attempt to `resetFormOwner()` here.